File: 0409_Longest_Palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        if not s:
            return ""

        letter_count = [0 for _ in range(128)]
        for i in range(len(s)):
            letter_count[ord(s[i])] += 1

        res = 0
        extra = False
        for i in range(len(letter_count)):
            res += letter_count[i] - (letter_count[i] % 2)
            if not extra and letter_count[i] % 2 == 1:
                res += 1
                extra = True

        return res

File: cpp/0409_Longest_Palindrome.cpp
#include <iostream>
#include <unordered_map>

using namespace std;

class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char, int>m;
        for (auto c : s) {
            m[c]++;
        }
        int ans = 0;
        bool extra = false;
        for (auto x : m) {
            int val = x.second;
            ans += val - (val % 2);
            if (!extra && val % 2 == 1) {
                ans++;
                extra = true;
            }
        }
        return ans;
    }
};